### DISCUSSION TITLE
Collapsible section after Heading

### DISCUSSION THREAD

**Discussion by JSaillok** [Daniele-rolli/Beaver-Notes#127] [Ideas]
Using Beaver Notes daily and finding new ways to edit my notes, I discovered a feature that other apps have but is not present here.
I am talking about the collapsible sections after headings. The reason I would like it is because a note can get very long and searching for a section can be difficult.
I tried to introduce this function myself but when I cloned the repository, I admit I got lost.
One way of approaching it is:

1. When the note data is rendered, check if an element is a heading element.
2. If it is a heading element, look at the next sibling element.
3. If the next sibling is a &lt;p&gt; &lt;/p&gt; tag, add a class like "collapse" to it.
4. In CSS style the "collapse" class to set the paragraph to be hidden by default:
`.collapse {  
  display: none;  
}`
5. In JavaScript a script to toggle the paragraph visibility on click of the heading:
`document.querySelectorAll('h1, h2, h3').forEach(heading => {  
   heading.addEventListener('click', () => {  
     const paragraph = heading.nextElementSibling;  
     if (paragraph.classList.contains('collapse')) {  
       paragraph.classList.toggle('show');  
     }  
   });  
});`  
`.collapse.show { display: block; }`

This will make all paragraphs immediately after a heading collapsed by default, but they can be toggled open when the heading is clicked.

**Comment by Daniele-rolli** [comment]
Thanks for sharing your suggestion! I appreciate your input. I must admit, collapsible headings are new to me, but I'll definitely add them to our to-do list. Also, I wanted to mention that Beaver has a handy search heading function designed to make note browsing easier, in case that's something you find helpful.

**Comment by Daniele-rolli** [comment]
It's been quite a while, but I'm proud to say that we managed to introduce collapsible headings in version 3.4.0 almost two weeks ago. We will are looking forward to improving this feature in the next release and also integrate it into our beaver pocket. Thanks again for proposing this feature, and I apologize for the delay in delivering it.
